common/log.h

```diff
@@ -154,7 +171,7 @@ inline std::string log_filename_generator_impl(const std::string & log_file_base
 //  #include "log.h"
 //
 #ifndef LOG_NO_TIMESTAMPS
-    #ifndef _WIN32
+    #if LOG_COMPAT == LOG_COMPAT_GNU
```

Comment:
Use `#ifndef _MSC_VER` here. Same for all below instances.

Reply:
I agree, I just found a segfault problem and I'm trying to determine if logs have anything to do with it

Reply:
Is that something I could reproduce myself?

Reply:
Basically, compile current commit with make (haven't tried it with cmake) under this https://github.com/skeeto/w64devkit (that's what the original issue involved)

And run `main` with no args.

I couldn't get a backtrace with gdb, but visual studio debugger, with the w64devkit build, shows a basic(hex offsets) backtrace pointing to somewhere in main.

I disabled logs completely in the build and it still segfaults.

There are other parts of the project where _WIN32 is used, and there are compile warnings because of it, under w64devkit, so I'm suspecting that's the cause, especially since i can't reproduce on Linux.

Reply:
I built this PR with MSYS2 as `mingw32-make LLAMA_DEBUG=1` and then ran `./main -m llama-7b.q4_0.gguf` with no issue. I'll have to try and install w64devkit instead.

Reply:
It's not related to the logging changes - I can reproduce it on commit 0728c5a8 (the first one I tried besides the tip of this PR).

Reply:
Hmm... I bisected it to commit 0c44427df10ee024b4e7ef7bfec56e993daff1db, which adds `-march=native`. Also, it's not 100% repeatable, but usually happens within 1 or 2 tries.

Reply:
Yup, for me it was like randomly, roughly half of the time

Reply:
I used cv2pdb to get some debug symbols, and the crash seems to be happening here:
https://github.com/ggerganov/llama.cpp/blob/8afe2280009ecbfc9de2c93b8f41283dc810609a/common/common.cpp#L723

Reply:
Based on the disassembly, something is going wrong before that function call:
```
    llama_model * model  = llama_load_model_from_file(params.model.c_str(), lparams);
00007FF7CBE59448  mov         rax,qword ptr [params]  
00007FF7CBE5944F  add         rax,0C8h  
00007FF7CBE59455  mov         rcx,rax  
00007FF7CBE59458  call        _M_range_check+0F70h (07FF7CBEA0340h)  
00007FF7CBE5945D  mov         rcx,rax  
00007FF7CBE59460  vmovdqu     ymm0,ymmword ptr [lparams]  
00007FF7CBE59465  vmovdqa     ymmword ptr [rbp-60h],ymm0  <-- segfault is here
00007FF7CBE5946A  vmovdqu     ymm0,ymmword ptr [rbp+10h]  
00007FF7CBE5946F  vmovdqa     ymmword ptr [rbp-40h],ymm0  
00007FF7CBE59474  lea         rax,[rbp-60h]  
00007FF7CBE59478  mov         rdx,rax  
00007FF7CBE5947B  call        llama_load_model_from_file (07FF7CBE4B9D7h)
```
rbp is 0x0000007CA91FE0D0, so I'm not sure where 0xFFFFFFFFFFFFFFFF comes from.

Reply:
I opened issue #2922 for this, so we don't clutter the review comments any further.